### ISSUE TITLE
Nested tags or components in Previews raise error

### ISSUE THREAD

**Issue by scottbarrow** [github/view_component#390]
### Steps to reproduce
Create a preview
add method from https://github.com/github/view_component#previewing-components example
```  ruby 
def with_content_block
    render(ExampleComponent.new) do
      tag.div do
        content_tag(:span, 'Hello')
      end
    end
  end
```

### Expected behavior
Preview to render an Example component html, with a div and span containing "Hello"

### Actual behavior
```
ActionView::Template::Error (undefined method `output_buffer=' for #<ExampleComponentPreview:0x00007fd5b0befab8>):
    1: <%= render(@render_args[:component], @render_args[:args], &@render_args[:block])%>
```

### System configuration
**Rails version**:
6.0.3
**Ruby version**:
2.6.5
**Gem version**:
2.11.1

N.b.
Also 
```  ruby 
def with_content_block
    render(ExampleComponent.new(title: 'parent') do
      render(ExampleComponent.new('child')
    end
  end
# html
<div>
  <%= @title %>
  <div><%= content %></div>
</div>

```
produces the parent component html but not the child in the block

**Comment by joelhawksley** [comment]
@scottbarrow thanks for the bug report! Would you be up for looking into a fix?

**Comment by scottbarrow** [comment]
@joelhawksley I will look into it a little, however I am new to the gem

**Comment by joelhawksley** [comment]
@scottbarrow Thanks! We all were new to the gem at some point ❤️ 

I'd recommend starting with an integration test. @juanmanuelramallo might be able to provide assistance here as well, as he has done a lot of the work on Previews.

**Comment by juanmanuelramallo** [comment]
Hey @scottbarrow some time ago I started working on this PR https://github.com/github/view_component/pull/353 which will enable us to write templates instead of just calling render in the method's body of preview objects. 

I haven't had enough time lately but I'll try to wrap it up shortly so that we support the use case mentioned in this issue.

Related: https://github.com/github/view_component/discussions/337

**Comment by avk** [comment]
Just hit this as well. Is there a workaround until something like #353 makes it in?

**Comment by scottbarrow** [comment]
In the second example I provided in my op, my workaround was to do
```ruby
 render(ExampleComponent.new(title: 'parent')) do |component|
       component.render(NestedComponent.new)
     end
```

**Comment by juanmanuelramallo** [comment]
#353 just got in. Would you mind taking a look at https://github.com/github/view_component#preview-templates and check if that solves this issue? I guess it does

**Comment by joelhawksley** [comment]
Marking this issue as closed per #353. Feel free to reopen if it does not resolve this concern ❤️
